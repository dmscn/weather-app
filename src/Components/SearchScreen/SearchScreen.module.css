.Container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: 100vh;
}

.TextField {
  width: 25vw;
}

.Button {
  margin-top: 3.5em;
}
