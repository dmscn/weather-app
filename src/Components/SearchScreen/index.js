export {default} from './SearchScreen';
export * from './SearchScreen';