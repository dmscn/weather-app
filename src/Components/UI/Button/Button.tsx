import React, { Component, ReactNode } from 'react';
import classNames from 'classnames';
import Button from '@material-ui/core/Button';

import styles from './Button.module.css';

export default class UIButton extends Component<any> {
  render() {
    const { className = "", children, ...rest } = this.props;
    return (
      <div className={classNames(styles.Container, this.props.className)}>
        <Button
          color="primary"
          variant="contained"
          {...rest}
        >
          {children}
        </Button>
      </div>
    )
  }
}
