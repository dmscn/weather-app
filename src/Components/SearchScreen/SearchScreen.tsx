import React, { Component } from 'react';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

import styles from './SearchScreen.module.css';

export interface Props {
  onSubmit: (location: string) => void;
}

export interface State {
  text: string,
}

export default class SearchScreen extends Component<Props, State> {
  state = {
    text: '',
  }

  handleOnChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
    this.setState({ text: event.target.value });
  }

  onSubmit = (): void => {
    this.props.onSubmit(this.state.text);
  }

  render(): JSX.Element {
    const { text } = this.state;
    return (
      <main className={styles.Container}>
        <TextField
          id="LocationInput"
          label="Location"
          placeholder="Location"
          value={text}
          onChange={this.handleOnChange}
          inputProps={{
            'aria-label': 'Search Box',
          }}
        />
        <Button
          variant="raised"
          onClick={this.onSubmit}
        >
          Search
        </Button>
      </main>
    )
  }
}
