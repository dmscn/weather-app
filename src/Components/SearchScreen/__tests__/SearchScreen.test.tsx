import React from 'react'
import { shallow } from "enzyme";
import Button from '@material-ui/core/Button';

import SearchScreen from "../SearchScreen";

describe('<SearchScreen />', () => {
  it('renders', () => {
    const wrapper = shallow(<SearchScreen onSubmit={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('changes text', () => {
    const wrapper: any = shallow(<SearchScreen onSubmit={() => {}}/>);
    const spy = jest.spyOn(wrapper.instance(), 'handleOnChange');
    const mEvent = { target: { name: 'Location', value: 'a'}};

    wrapper.find('#LocationInput').simulate('change', mEvent);
    expect(wrapper.state()).toEqual({text: 'a'});
  })

  it('submits when button clicked', () => {
    const mSubmitFunction = jest.fn();
    const wrapper = shallow(<SearchScreen onSubmit={mSubmitFunction}/>);

    wrapper.find(Button).simulate('click');
    expect(mSubmitFunction).toHaveBeenCalledWith('');
  })
});