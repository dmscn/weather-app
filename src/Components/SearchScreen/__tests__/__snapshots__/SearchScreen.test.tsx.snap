// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchScreen /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchScreen
    onSubmit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TextField
          id="LocationInput"
          inputProps={
            Object {
              "aria-label": "Search Box",
            }
          }
          label="Location"
          onChange={[Function]}
          placeholder="Location"
          required={false}
          select={false}
          value=""
          variant="standard"
        />,
        <WithStyles(Button)
          onClick={[Function]}
          variant="raised"
        >
          Search
        </WithStyles(Button)>,
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "LocationInput",
          "inputProps": Object {
            "aria-label": "Search Box",
          },
          "label": "Location",
          "onChange": [Function],
          "placeholder": "Location",
          "required": false,
          "select": false,
          "value": "",
          "variant": "standard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Search",
          "onClick": [Function],
          "variant": "raised",
        },
        "ref": null,
        "rendered": "Search",
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TextField
            id="LocationInput"
            inputProps={
              Object {
                "aria-label": "Search Box",
              }
            }
            label="Location"
            onChange={[Function]}
            placeholder="Location"
            required={false}
            select={false}
            value=""
            variant="standard"
          />,
          <WithStyles(Button)
            onClick={[Function]}
            variant="raised"
          >
            Search
          </WithStyles(Button)>,
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "LocationInput",
            "inputProps": Object {
              "aria-label": "Search Box",
            },
            "label": "Location",
            "onChange": [Function],
            "placeholder": "Location",
            "required": false,
            "select": false,
            "value": "",
            "variant": "standard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Search",
            "onClick": [Function],
            "variant": "raised",
          },
          "ref": null,
          "rendered": "Search",
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
